### PR TITLE
update ginko to v2, CAPI to 1.3.x, go to 1.19 and certmanager to v1.10.0

### PR DESCRIPTION
**Pull request by cpanato**
**What type of PR is this?**

/kind feature
/kind cleanup

**What this PR does / why we need it**:

- update ginko to v2, CAPI to 1.3.x, go to 1.19 and certmanager to v1.10.0

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
update ginko to v2, CAPI to 1.3.x, go to 1.19 and certmanager to v1.10.0
```

### CODE REVIEW

**Comment by cpanato** [comment]
This will require update the prow jobs

**Comment by cpanato** [comment]
prow job update: https://github.com/kubernetes/test-infra/pull/28115

**Comment by cpanato** [comment]
/assign @dims @richardcase @sbueringer @fabriziopandini

this is the one :)
/hold

File: Tiltfile [hunk 1]
```diff
@@ -18,8 +18,8 @@ settings = {
     "deploy_cert_manager": True,
     "preload_images_for_kind": True,
     "kind_cluster_name": "capg",
-    "capi_version": "v1.2.6",
-    "cert_manager_version": "v1.1.0",
+    "capi_version": "v1.3.0-rc.1",
+    "cert_manager_version": "v1.10.0",
```
Comment (sbueringer) [inline Tiltfile:22]: Just if you want to pick it up, there was a v1.10.1 release recently

File: go.mod [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
 
-go 1.18
+go 1.19
 
 require (
 	github.com/GoogleCloudPlatform/k8s-cloud-provider v1.20.0
 	github.com/google/go-cmp v0.5.9
-	github.com/onsi/ginkgo v1.16.5
-	github.com/onsi/gomega v1.22.1
+	github.com/onsi/ginkgo/v2 v2.5.0
```
Comment (sbueringer) [inline go.mod:8]: Similar > https://github.com/onsi/ginkgo/releases/tag/v2.5.1

File: scripts/ci-conformance.sh [hunk 3]
```diff
@@ -45,8 +45,8 @@ export GCP_NETWORK_NAME=${GCP_NETWORK_NAME:-"${TEST_NAME}-mynetwork"}
 GCP_B64ENCODED_CREDENTIALS=$(base64 -w0 "$GOOGLE_APPLICATION_CREDENTIALS")
 export GCP_B64ENCODED_CREDENTIALS
 export KUBERNETES_MAJOR_VERSION="1"
-export KUBERNETES_MINOR_VERSION="23"
-export KUBERNETES_PATCH_VERSION="8"
+export KUBERNETES_MINOR_VERSION="24"
```
Comment (sbueringer) [inline scripts/ci-conformance.sh:48]: No idea, but could/should this be 25? (similar in other places)

**Review by sbueringer — COMMENTED** [review]
Looks good to me just some optional nits

/lgtm

File: Tiltfile (hunk 1 above)
Comment (cpanato) [reply Tiltfile:22]: cool, thanks for your review

File: scripts/ci-conformance.sh (hunk 3 above)
Comment (cpanato) [reply scripts/ci-conformance.sh:48]: ok

**Comment by sbueringer** [comment]
Thx! 

/lgtm

**Comment by cpanato** [comment]
i will fix the `pull-cluster-api-provider-gcp-e2e-workload-upgrade` in a follow up, this is not required for now

/unhold

**Pull request merged**